nodes/liveplot

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+"""
+plots ros superpackets in 3D
+"""
+import matplotlib.pyplot as plt
+import mpl_toolkits.mplot3d.axes3d as p3
+import matplotlib.animation as animation
+import roslib
+roslib.load_manifest('strawlab_freeflight_experiments')
+import rospy
+import ros_flydra.msg
+import argparse
+import collections
+import itertools
+import math
+
+
+class RosHandler():
+    '''
+    gets the flydra_mainbrain_super_packet from ros
+    '''
+    def __init__(self):
+        self._super_packet = None
+        rospy.Subscriber("/flydra_mainbrain/super_packets",
+                         ros_flydra.msg.flydra_mainbrain_super_packet,
+                         self._on_target_move)
+
+    def _on_target_move(self, msg):
+        self._super_packet = msg.packets[0]
+
+    def get_super_packet(self):
+        return self._super_packet
+
+
+class DataHandler():
+    '''
+    we store xzy for every objid with this class to a circular buffer
+    new objIds are added, old ones that are not updated for a periode of time
+    are deleted
+    '''
+    def __init__(self, buffer_length=1000, dead_after_secs=30):
+        self.obj = {}  # memory
+        self.buffer_length = buffer_length
+        self.dead_after_secs = dead_after_secs
+        self._super_packet = None
+        rospy.Subscriber("/flydra_mainbrain/super_packets",
+                         ros_flydra.msg.flydra_mainbrain_super_packet,
+                         self._on_target_move)
+
+    def _on_target_move(self, msg):
+        self._super_packet = msg.packets[0]
+
+    def aquire(self):
+        '''
+        update data of obj
+        '''
+        new_dataset = self._super_packet
+        if new_dataset != None:  # is none if nothing comes from ros
+            teimestamp = new_dataset.acquire_stamp.secs
+            objects = new_dataset.objects
+            # add data from objects
+            for obj in objects:
+                # if new obj
+                if obj.obj_id not in self.obj:  # add ob to dict if not existing
+                    self.obj[obj.obj_id] = {}
+                    self.obj[obj.obj_id]['x'] = collections.deque(
+                                        [obj.position.x] * self.buffer_length,
```

Comment:
fair enough although I understood that matplotlib didn't plot NaNs, so if the buffer was ````0.5, nan, nan, nan```` then it would only plot the real values. as the next packet arrived it would push out one nan ````0.6,0.5,nan,nan````

etc